app/cafe.py

```diff
@@ -0,0 +1,19 @@
+from app.errors import NotVaccinatedError, \
+                        OutdatedVaccineError, \
+                        NotWearingMaskError
+import datetime
+
+
+class Cafe:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def visit_cafe(self, vstr: dict) -> str:
+        if "vaccine" not in vstr.keys() or vstr["vaccine"] == None:
+            raise NotVaccinatedError("not vaccinated")
+        elif vstr["vaccine"]["expiration_date"] < datetime.date.today():
+            raise OutdatedVaccineError("outdated")
+        elif not vstr["wearing_a_mask"]:
+            raise NotWearingMaskError("not wearing mask")
+        else:
+            return "Welcome to " + self.name
```

Comment:
Several changes can be made here:

- [ ] When checking if something is `None`, use `is` instead of `==`
- [ ] The error message can be improved using an f-string to identify which visitor is not allowed to enter the cafe. This is useful if in the future you want to log function errors
- [ ] The `else` before the `return` statement can be removed as it is redundant
- [ ] After each `raise` statement, you can insert a line break to improve readability

The final code should look like this:

```suggestion
        if "vaccine" not in visitor.keys() or visitor["vaccine"] is None:
            raise NotVaccinatedError(f"Visitor {visitor["name"]} is not vaccinated.")

        elif visitor["vaccine"]["expiration_date"] < datetime.date.today():
            raise OutdatedVaccineError(f"Visitor {visitor["name"]} vacine is outdated.")

        elif not visitor["wearing_a_mask"]:
            raise NotWearingMaskError(f"Visitor {visitor["name"]} is not wearing mask")

        return "Welcome to " + self.name
```